src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStoreV1.cs

```diff
@@ -62,7 +63,8 @@ public virtual async Task<long> BeginCreateInstanceIndexAsync(int partitionKey,
 
             try
             {
-                return (long)(await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken));
+                long watermark = (long)await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
+                return new InstanceStorageKey { Watermark = watermark, InstanceKey = null };
```

Comment:
I appreciate being so **explicit**, but you can also just skip `InstanceKey` if you wanted